Python/sqlmlutils/sqlqueryexecutor.py

```diff
@@ -38,83 +35,81 @@ class SQLQueryExecutor:
     def __init__(self, connection: ConnectionInfo):
         self._connection = connection
 
-    def execute(self, builder: SQLBuilder, out_file=None, getResults=True):
+    def execute(self, builder: SQLBuilder, out_file=None):
+        return self.execute_query(builder.base_script, builder.params, out_file=out_file)
+
+    def execute_query(self, query, params, out_file=None):
+        df = DataFrame()
+        output_params = None
+
         try:
             if out_file is not None:
                 with open(out_file,"a") as f:
-                    if builder.params is not None:
-                        script = builder.base_script.replace("%s", "N'%s'")
-                        f.write(script % builder.params)
+                    if params is not None:
+                        script = query.replace("?", "N'%s'")

```

Comment:
params may not contain only strings. The replace is probably not going to work in that case. i was thinking to dump the query and the parameters separately. it may need more formatting for binaries.

Reply:
This is not input or output params, this is **builder** params, or a SQL query. Builder is an abstract class in this project that returns a base query and a set of params to execute with the query, and we control both parts, so we can guarantee it is a string.

With that said, the bytearray that is returned by the PackageSqlBuilder class is printed in an unusable way, so I will detect that and convert it to hex here.